app/models/user.rb

```diff
@@ -274,7 +274,8 @@ def create_classrooms!
   def generate_student_username_if_absent
     return if not student?
     return if username.present?
-    generate_username
+    classroom_id = classrooms.any? ? classrooms.first.id : nil
```

Comment:
if they create their own account, they don't instantly have a classroom to add to the username